src/core/audio/AudioJack.cpp

```diff
@@ -405,6 +430,21 @@ int AudioJack::processCallback( jack_nframes_t _nframes, void * _udata )
 		}
 	}
 
+
+	m_inBuffer.reset (new sampleFrame[_nframes]);
+
+	const float gain = mixer()->masterGain();
+	for( int c = 0; c < channels(); ++c ) {
+		jack_default_audio_sample_t *channel_buffer = m_tempInBufs[c];
+
+		for( jack_nframes_t frame = 0; frame < _nframes; ++frame )
+		{
+			m_inBuffer[frame][c] = channel_buffer[frame] / gain;
+		}
+	}
+
+	mixer()->pushInputFrames (m_inBuffer.get (), _nframes);
+
```

Comment:
This whole logic should be run if and only if JACK is running(in other words, `m_stopped` is false). Otherwise, it will cause deadlock after you export audio with sample rate different from JACK's processing sample rate.
LMMS doesn't deactivate the client when the `AudioJack` interface stops processing.

Reply:
Just to be sure, where is the deadlock actually happening?

Reply:
JACK callback calls `requestChangeInModel()` which waits for mixer, before the mixer starts. The mixer can't be started because the function blocks starting the mixer thread.
Relevant parts of backtrace:
```gdb
Thread 10 (Thread 0x7fffd7887700 (LWP 27891)):
#0  pthread_cond_wait@@GLIBC_2.3.2 ()
    at ../sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S:185
#1  0x00007ffff48baa5b in QWaitCondition::wait(QMutex*, unsigned long) ()
   from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#2  0x00000000005322ed in Mixer::requestChangeInModel() ()
#3  0x000000000053239f in Mixer::pushInputFrames(float const (*) [2], int, bool) ()
#4  0x000000000057a9ef in AudioJack::processCallback(unsigned int, void*) ()


Thread 1 (Thread 0x7ffff7ee0900 (LWP 27879)):
#0  syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
#1  0x00007ffff48b18b8 in QBasicMutex::lockInternal() ()
   from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#2  0x00007ffff48b19ad in QMutex::lock() ()
   from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#3  0x0000000000532299 in Mixer::requestChangeInModel() ()
#4  0x00000000005572b2 in SampleBuffer::update(bool) ()
#5  0x00007ffff4ac8d2a in QMetaObject::activate(QObject*, int, int, void**) ()
   from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#6  0x00000000005336bb in Mixer::restoreAudioDevice() ()
```
FYI `QMetaObject::activate` is called here: https://github.com/LMMS/lmms/blob/03e9889605fbe729bc479412b9d1767f5b39103d/src/core/Mixer.cpp#L641

Reply:
Fixed.